src/main/scala/nl.knaw.dans.easy.bagstore/BagStoreEnum.scala

```diff
@@ -22,22 +22,25 @@ import scala.util.Try
 
 trait BagStoreEnum { this: BagFacadeComponent with BagStoreContext =>
 
-  def enumBags(includeVisible: Boolean = true, includeHidden: Boolean = false): Try[Stream[BagId]] = Try {
-    Files.walk(baseDir, uuidPathComponentSizes.size).iterator().asScala.toStream
-      .map(baseDir.relativize)
-      .withFilter(_.getNameCount == uuidPathComponentSizes.size)
-      .map(p => fromLocation(baseDir.resolve(p)).flatMap(_.toBagId).get) // TODO: is there a better way to fail fast ?
-      .filter(bagId => {
+  // TODO: support huge numbers of bags. (The stream should then probably NOT be converted in to a List anymore!)
```

Comment:
Good point, indeed we should have this either as a `Stream` or as an `Observable`. In the former case we would have a return type `Try[Stream[Try[BagId]]]`, the latter would have return type `Observable[BagId]`.

**Explanation for the former return type:** Something can fail during the walk through the filesystem, so there is the outer `Try`. The `Stream` replaces the current `Seq`. Since we want to fail fast (see TODO in the code below) and want to signal that `Try` downstream, you need another `Try` around every `BagId`.

**Good point to note here:** Even though we would possibly return a stream of some kind here, it is still returned as kind of list in the REST interface. Is there a way to stream the results to the user via REST? I don't mean here to do paging, but proper streaming (push based, as it ought to be). Or do we need another protocol for that?